docker-compose.yaml

```diff
@@ -11,23 +13,43 @@ services:
       - 3000:3000
     environment:
       - DATABASE_URL="postgresql://postgres:postgres@postgres:5432/merch"
-    networks:
-      - app_network
-
+    depends_on:
+      - postgres
+  web-dev:
+    container_name: web-dev
+    build:
+      context: .
+      dockerfile: Dockerfile
+    volumes:
+      - .:/app
+      # NOTE(luisd): this is the main downsize of using volumes and pnpm
+      # pnpm requires a store, but because it's inside of a container we can't access
+      # the default (if any) store, so we override the root node_modules and the local store
+      - /app/node_modules
+      - /app/.pnpm-store
+    depends_on:
+      - caddy
+      - postgres
+    # FIXME(luisd): when UPDigital gives us a localhost redirect_uri we can change it back
+    # for now, we use caddy for ssl termination (and a script to redirect 
+    # ni.fe.up.pt to localhost while it's running)
+    #ports:
+    #  - 3000:3000 
+  # caddy is also necessary while we sort the redirect_uri out
+  caddy:
+    image: docker.io/caddy
+    ports:
+      - 80:80
+      - 443:443
+    volumes:
+      - ./Caddyfile:/etc/caddy/Caddyfile
+      - ./localhost.cert:/localhost.cert
+      - ./localhost.key:/localhost.key
   postgres:
-    image: postgres:16
+    image: docker.io/postgres:16
     restart: always
     environment:
       POSTGRES_USER: postgres
       POSTGRES_PASSWORD: postgres
-    ports:
-      - "5432:5432"
```

Comment:
Having this is useful for running Prisma Studio and debugging stuff.

I understand that in prod this is unwanted and therefore I think we should have a docker-compose.prod.yml and a docker-conpose.dev.yml